aioftp/client.py

```diff
@@ -338,6 +341,188 @@ def parse_directory_response(self, s):
 
         return pathlib.PurePosixPath(directory)
 
+    @staticmethod
+    def parse_unix_mode(s):
+        """
+        Parsing unix mode strings("rwxr-x--t") into
+        hexacimal notation
+
+        :param s: mode string
+        :type s: :py:class:`str`
+
+        :return mode:
+        :rtype: :py:class:`int`
+        """
+        def parse_rw(s):
+            mode = 0
+
+            if s[0] == 'r':
+
+                mode |= 4
+
+            elif s[0] != '-':
+
+                raise ValueError
+
+            if s[1] == 'w':
+
+                mode |= 2
+
+            elif s[1] != '-':
+
+                raise ValueError
+
+            return mode
+
+        mode = 0
+        mode |= parse_rw(s[0:2]) << 6
+        mode |= parse_rw(s[3:5]) << 3
+        mode |= parse_rw(s[6:8])
+
+        if s[2] == 's':
```

Comment:
Maybe some dictionaries instead of imperative way? All bits are just `symbol`-`number` pairs, it looks like it is good ground for declarative description.